### ISSUE TITLE
confidence loss

### ISSUE THREAD

**Issue by raphaellndr** [david8862/keras-YOLOv3-model-set#124]
Hello,
I'm currently trying to understand the yolo3_loss function.
When not using focal loss, the confidence loss is as follow :

confidence_loss = object_mask * K.binary_crossentropy(true_objectness_probs, raw_pred[...,4:5], from_logits=True) +
                (1-object_mask) * K.binary_crossentropy(object_mask, raw_pred[...,4:5], from_logits=True) * ignore_mask

In the original article, there is a mention of a lambda_noobj coefficient=0.5, to avoid too much focus on no-object prediction, is there something similar in this code ? I can't find it.
Moreover, I don't understand how the model don't tend to only produce negative prediction ? (even with the coefficient) since the model needs to produce far more negative than positive prediction.
Thanks in advance

**Comment by david8862** [comment]
@raphaellndr  this part is mainly inherited from qqwweee's original repo. From my understanding, the negative sample number in objectness loss here is also controlled by "ignore_mask", with which we only pick the "pure background" in grid cells to calculate the noobj loss, so that to have pos/neg balance.

**Comment by raphaellndr** [comment]
Hello @david8862 , thank you for your quick response.
the ignore_mask remove boxes with GT_iou>treshold (default=0.5), but it's still very imbalance. I will check the qqwweee's repository to see if there is more informations.
Regards

**Comment by ShihuaiXu** [comment]
Sir, have you found any solution to confirm this question?
@raphaellndr
